lib/ecto/repo/schema.ex

```diff
@@ -552,14 +558,23 @@ defmodule Ecto.Repo.Schema do
           apply(mod, on_delete, [reflection, changeset.data, name, tuplet])
         end
 
-        schema_meta = metadata(struct, schema.__schema__(:autogenerate_id), opts)
+        autogen_id = schema.__schema__(:autogenerate_id)
+        schema_meta = metadata(struct, autogen_id, opts)
+        {_changes, _cast_extra, _dump_extra, return_types, return_sources} =
+          autogenerate_id(autogen_id, changeset.changes, return_types, return_sources, adapter)
```

Comment:
I don't believe we need to call this function, as we should not autogenerate keys on deletions. Is there a reason why you call it?

Reply:
I didn't see your comment before I posted it below: https://github.com/elixir-ecto/ecto/pull/4277#issuecomment-1721278859

In short, I followed the other tests, and I think I might have gotten caught up by the ID not being included:

https://github.com/elixir-ecto/ecto/blob/master/test/ecto/repo_test.exs#L506L529

In my mind, it makes sense always to include the primary keys when only the `:parent_id` key has been selected since it might not be enough information to determine which row has been deleted.

FWIW, I'm not opposed to reverting this, but I look forward to hearing your decision.